src/dbally/views/freeform/text2sql/_autodiscovery.py

```diff
@@ -0,0 +1,328 @@
+        """
+        Set the whitelist of tables to include in the auto-discovery.
+
+        Args:
+            whitelist: List of table names to include in the auto-discovery.
+
+        Returns:
+            _AutoDiscoveryBuilder: The builder instance.
+
+        Raises:
+            ValueError: If both a whitelist and a blacklist are set.
+        """
+        if self._blacklist is not None:
+            raise ValueError("Cannot have both a blacklist and a whitelist")
+
+        self._whitelist = whitelist
+        return self
+
+    def extract_description_from_comments(self) -> Self:
+        """
+        Use the comments field in the database as a source for the table descriptions.
+
+        Returns:
+            _AutoDiscoveryBuilder: The builder instance.
+        """
+        self._description_extraction = _DescriptionExtractionStrategy()
+        return self
+
+    async def discover(self) -> Text2SQLConfig:
+        """
+        Discover the tables in the database and return the configuration object.
+
+        Returns:
+            Text2SQLConfig: The configuration object for the text2sql freeform view.
+        """
+        return await _Text2SQLAutoDiscovery(
+            llm_client=self._llm_client,
+            engine=self._engine,
+            whitelist=self._whitelist,
+            blacklist=self._blacklist,
+            description_extraction=self._description_extraction,
+            similarity_enabled=self._similarity_enabled,
+        ).discover()
+
+
+class _AutoDiscoveryBuilderWithLLM(_AutoDiscoveryBuilderBase):
+    def generate_description_by_llm(self, example_rows_cnt: int = 5) -> Self:
+        """
+        Use LLM to generate descriptions for the tables.
+        The descriptions are generated based on the table DDL and a configured count of example rows.
+
+        Args:
+            example_rows_cnt: The number of example rows to use for generating the description.
+
+        Returns:
+            _AutoDiscoveryBuilder: The builder instance.
+        """
+        self._description_extraction = _LLMSummaryDescriptionExtraction(example_rows_cnt)
+        return self
+
+
+class _AutoDiscoveryBuilder(_AutoDiscoveryBuilderBase):
+    def use_llm(self, llm_client: LLMClient) -> _AutoDiscoveryBuilderWithLLM:
+        """
+        Set the LLM client to use for generating descriptions.
+
+        Args:
+            llm_client: The LLM client to use for generating descriptions.
+
+        Returns:
+            _AutoDiscoveryBuilderWithLLM: The builder instance.
+        """
+        return _AutoDiscoveryBuilderWithLLM(
+            engine=self._engine,
+            whitelist=self._whitelist,
+            blacklist=self._blacklist,
+            description_extraction=self._description_extraction,
+            similarity_enabled=self._similarity_enabled,
+            llm_client=llm_client,
+        )
+
+
+def configure_text2sql_auto_discovery(engine: Engine) -> _AutoDiscoveryBuilder:
```

Comment:
I'm not a fan of returning instances of private classes from public functions - i.e., making a private type part of the public interface. Since the users will be handling the object they may want to reference it's type. So in my mind `AutoDiscoveryBuilder` should be public as well.

Reply:
you are right, fixed